bindings/rascal/utils/filter.py

```diff
@@ -0,0 +1,568 @@
+            sps = list(n_select.keys())
+            # get various info from the structures about the center atom species and indexing
+            (
+                strides_by_sp,
+                global_counter,
+                map_by_manager,
+                indices_by_sp,
+            ) = get_index_mappings_sample_per_species(managers, sps)
+            selected_ids_by_sp = {
+                key: np.sort(val[: n_select[key]])
+                for key, val in self.selected_sample_ids_by_sp.items()
+            }
+            self.selected_ids = (
+                convert_selected_global_index2perstructure_index_per_species(
+                    managers,
+                    selected_ids_by_sp,
+                    strides_by_sp,
+                    map_by_manager,
+                    sps,
+                )
+            )
+            # return self.selected_ids
+            # build the pseudo points
+            pseudo_points = SparsePoints(self._representation)
+            pseudo_points.extend(managers, self.selected_ids)
+            return pseudo_points
+
+        elif self.act_on == "sample":
+
+            selected_ids_global = np.sort(self.selected_sample_ids[:n_select])
+            strides, _, map_by_manager = get_index_mappings_sample(managers)
+            self.selected_ids = convert_selected_global_index2perstructure_index(
+                managers, selected_ids_global, strides, map_by_manager
+            )
+            return self.selected_ids
+            # SparsePoints is not compatible with a non center atom species
+            # dependant sparse points
+            # # build the pseudo points
+            # pseudo_points = SparsePoints(self._representation)
+            # pseudo_points.extend(managers, self.selected_ids)
+            # return pseudo_points
+
+        elif self.act_on == "feature":
+            feat_idx2coeff_idx = self._representation.get_feature_index_mapping(
+                managers
+            )
+            self.selected_ids = {key: [] for key in feat_idx2coeff_idx[0].keys()}
+            selected_ids_sorting = np.argsort(
+                self.selected_feature_ids_global[:n_select]
+            )
+            selected_feature_ids = self.selected_feature_ids_global[
+                selected_ids_sorting
+            ]
+            for idx in selected_feature_ids:
+                coef_idx = feat_idx2coeff_idx[idx]
+                for key in self.selected_ids.keys():
+                    self.selected_ids[key].append(int(coef_idx[key]))
+            # keep the global indices and ordering for ease of use
+            self.selected_ids[
+                "selected_feature_ids_global"
+            ] = selected_feature_ids.tolist()
+            self.selected_ids[
+                "selected_feature_ids_global_selection_ordering"
+            ] = selected_ids_sorting.tolist()
+            self.selected_ids = dict(coefficient_subselection=self.selected_ids)
+            return self.selected_ids
```

Comment:
In this case it makes no sense to return a `SparsePoints`, since the selection is done on features. What's returned here is a dictionary you can use to update the hypers and construct a new representation that computes the features sparsely (see the new example notebook for usage). But the docstring should definitely be updated to reflect this.